articles/200_migration_guide_from_ros1.md

```diff
@@ -0,0 +1,153 @@
+
+* Replace the invocation of `add_message_files`, `add_service_files` and `generate_messages` with [rosidl_generate_interfaces](https://github.com/ros2/rosidl/blob/master/rosidl_cmake/cmake/rosidl_generate_interfaces.cmake).
+
+* Remove any occurrences of the *devel space*.
+  Related CMake variables like `CATKIN_DEVEL_PREFIX` do not exist anymore.
+
+* Replace `CATKIN_*_DESTINATION` variables with their actual paths according to the [FHS](http://www.pathname.com/fhs/):
+
+  * `CATKIN_GLOBAL_BIN_DESTINATION`: `bin`
+  * `CATKIN_GLOBAL_INCLUDE_DESTINATION`: `include`
+  * `CATKIN_GLOBAL_LIB_DESTINATION`: `lib`
+  * `CATKIN_GLOBAL_LIBEXEC_DESTINATION`: `lib`
+  * `CATKIN_GLOBAL_SHARE_DESTINATION`: `share`
+
+  * `CATKIN_PACKAGE_BIN_DESTINATION`: `lib/${PROJECT_NAME}`
+  * `CATKIN_PACKAGE_INCLUDE_DESTINATION`: `include/${PROJECT_NAME}`
+  * `CATKIN_PACKAGE_LIB_DESTINATION`: `lib`
+  * `CATKIN_PACKAGE_SHARE_DESTINATION`: `share/${PROJECT_NAME}`
+
+* Replace ROS 1 dependencies like `roscpp` with ROS 2 dependencies like `rclcpp`.
+  Using `rclcpp` will also require to link explicitly against one rmw implementation, commonly the default is provided by the [rmw_implementation](https://github.com/ros2/rmw_implementation/tree/master/rmw_implementation) package.
+
+
+#### Continue to use `catkin` in CMake
+
+ROS 2 uses ament as the build system but for backward compatibility ROS 2 has a package called `catkin` which provides almost the same API as catkin in ROS 1.
+In order to use this backward compatibility API the `CMakeLists.txt` must only be updated to call the function `catkin_ament_package()` *after* all targets.
+
+<div class="alert alert-warning" markdown="1">
+NOTE: This has not been implemented yet and is only an idea at the moment.
+Due to the amount of changes related to dependencies it has not yet been decided if this compatibility API is useful enough to justify the effort.
+</div>
+
+
+### Update source code
+
+#### Messages and services
+
+The namespace of ROS 2 messages and services uses a subnamespace (`msg` or `srv`) after the package name.
+Therefore an include looks like: `#include <my_interfaces/msg/MyMessage.hpp>`.
+The C++ type is then named: `my_interfaces::msg::MyMessage`.
+
+Shared pointer types are provided as typedefs within the message structs: `my_interfaces::msg::MyMessage::SharedPtr` as well as `my_interfaces::msg::MyMessage::ConstSharedPtr`.
+
+For more details please see the article about the [generated C++ interfaces](http://design.ros2.org/articles/generated_interfaces_cpp.html).
+
+
+#### ROS client library
+
+<div class="alert alert-warning" markdown="1">
+NOTE: to be writtten
+</div>
+
+
+## Launch files
+
+While launch files in ROS 1 are specified using [.xml](http://wiki.ros.org/roslaunch/XML) files ROS 2 uses Python scripts to enable more flexibility (see [launch package](https://github.com/ros2/launch/tree/master/launch)).
```

Comment:
I'm not sure we should fill this part out yet. We don't even have a design document which states what the launch solution will be like or why it will be like that. I don't think it's out of the realm of possibility to also have a static file format similar to roslaunch in addition to a Python interface. I'm not advocating for one thing or the other, I just want to point out that, imo, there is a lot of uncertainty around this part of ROS 2 right now.